### DISCUSSION TITLE
Typescript error: Type 'undefined' cannot be used as an index type.ts(2538)

### DISCUSSION THREAD

**Discussion by deadcoder0904** [nextauthjs/next-auth#2886] [Help] [answered]
I was following along https://spacejelly.dev/posts/how-to-make-twitter-api-requests-with-nextauth-js-session-tokens/#step-1-accessing-twitter-oauth-tokens-in-a-nextauthjs-session & found this chunk of code written in JS:

```ts
callbacks: {
    async jwt({ token, user, account = {}, profile, isNewUser }) {
      if (account.provider && !token[account.provider]) {
        token[account.provider] = {}
      }

      if (account.access_token) {
        token[account.provider].accessToken = account.access_token
      }

      if (account.refresh_token) {
        token[account.provider].refreshToken = account.refresh_token
      }

      return token
    },
  },
```

On the 2nd & 3rd if statements, I get an error on `account.provider`:

> Type 'undefined' cannot be used as an index type.ts(2538)

I've tried all general methods to fix it using TS like `as string` or `!` but I can't manage to find a fix.

What's the solution for this?

**Comment by balazsorban44** [comment, marked as answer]
just simplify to one if statement with `if(account)`

This should work:

```ts
async jwt({ token, account }) {
  if (account) {
    token[account.provider] = {
      accessToken: account.access_token,
      refreshToken: account.refresh_token,
    }
  }

  return token
}
```

Furthermore, to make TS happy, check out the docs at: https://next-auth.js.org/getting-started/typescript#submodules


```ts
declare module "next-auth/jwt" {
  interface JWT {
    [key: string]: {
      accessToken: string
      refreshToken: string
    }
  }
}
```

**Reply by deadcoder0904** [reply]
Thanks @balazsorban44 that worked. I saw the TS docs but didn't knew what to put inside.
